### PR TITLE
[#1547] Style: change if statements to use ternary operator

### LINKED ISSUE #1547: Use ternary operator to make the code style cleaner
for future reference, this may be a good place to use the ternary operator.

```java
return currentAnnotatedAuthor.isPresent() ? Optional.empty() : Optional.of(Author.UNKNOWN_AUTHOR);
```

_Originally posted by @damithc in https://github.com/reposense/RepoSense/pull/1520#discussion_r636888963_

```
As a language feature, we can use the ternary operator where appropriate to make the code style cleaner.
```

### PR DESCRIPTION
**Pull request by yhtMinceraft1010X**
Resolves #1547.

Proposed commit message:

```
Some of the if statements can be simplified to use the ternary
operator for better readability. This also makes the code look
cleaner.

Let's change these if statements into the ternary operator.
```

### CODE REVIEW

File: src/main/java/reposense/authorship/FileInfoExtractor.java [hunk 1]
```diff
@@ -66,11 +66,8 @@
         String lastCommitHash = GitRevList.getCommitHashBeforeDate(
                 config.getRepoRoot(), config.getBranch(), config.getSinceDate());
 
-        if (!lastCommitHash.isEmpty()) {
-            fileInfos = getEditedFileInfos(config, lastCommitHash);
-        } else {
-            fileInfos = getAllFileInfo(config, false);
-        }
+        fileInfos = lastCommitHash.isEmpty() ? getAllFileInfo(config, false)
+                : getEditedFileInfos(config, lastCommitHash);
```
Comment (HCY123902) [inline src/main/java/reposense/authorship/FileInfoExtractor.java:71]: I think line 194 to line 199 also needs to be changed to use ternary operator right

**Review by HCY123902 — COMMENTED** [review]
I think line 194 to line 199 also needs to be changed to binary

**Comment by HCY123902** [comment]
Thanks for the contribution. Is it possible that you can also help to include the rule for ternary operator usage into the programming style standard in the RepoSense Developer Guide? The Developer Guide is located in `docs/dg`

**Review by HCY123902 — APPROVED** [review]

File: src/main/java/reposense/authorship/FileInfoExtractor.java [hunk 2]
```diff
@@ -194,12 +191,8 @@ private static void setLinesToTrack(FileInfo fileInfo, String fileDiffResult) {
...
                 continue;
             }
-            if (isBinaryFiles) {
-                // skip LineInfo analysis
-                fileInfos.add(new FileInfo(relativePath.toString()));
-            } else {
-                fileInfos.add(generateFileInfo(config.getRepoRoot(), relativePath.toString()));
-            }
+            fileInfos.add(isBinaryFiles ? new FileInfo(relativePath.toString())
+                    : generateFileInfo(config.getRepoRoot(), relativePath.toString()));
```
Comment (dcshzj) [inline src/main/java/reposense/authorship/FileInfoExtractor.java:195]: See earlier comment.

File: src/main/java/reposense/report/RepoCloner.java [hunk 3]
```diff
@@ -93,11 +93,8 @@ public void cloneBare(RepoConfiguration config) {
             } catch (CommitNotFoundException e) {
                 sinceDate = null;
             }
-            if (sinceDate != null) {
-                isCurrentRepoCloned = spawnShallowCloneProcess(config, sinceDate);
-            } else {
-                isCurrentRepoCloned = spawnCloneProcess(config);
-            }
+            isCurrentRepoCloned = sinceDate != null ? spawnShallowCloneProcess(config, sinceDate)
+                    : spawnCloneProcess(config);
```
Comment (dcshzj) [inline src/main/java/reposense/report/RepoCloner.java:97]: See earlier comment.

File: src/main/java/reposense/util/StringsUtil.java [hunk 4]
```diff
@@ -39,10 +39,7 @@ public static String addQuote(String original) {
      * Removes quotes at the start and end of {@code original}, if exists.
      */
     public static String removeQuote(String original) {
-        if (original.startsWith("\"") && original.endsWith("\"")) {
-            return original.substring(1, original.length() - 1);
-        }
-
-        return original;
+        return original.startsWith("\"") && original.endsWith("\"")
+                ? original.substring(1, original.length() - 1) : original;
```
Comment (dcshzj) [inline src/main/java/reposense/util/StringsUtil.java:43]: See earlier comment.

**Review by dcshzj — APPROVED** [review]
LGTM!

**Pull request merged**